### ISSUE TITLE
Selecting different QR code from code.

### ISSUE THREAD

**Issue by brianchasalow** [googlevr/gvr-unity-sdk#287]
We would like to have a settings menu in our app so that we can quickly let users choose between a few common predefined settings- Cardboard v1, v2, v2.2, etc. 

As far as I am aware, (last I checked/inquired about this) the only way to change the QR code is to scan it, or alternatively we can set a default so that people who haven't ever scanned a code can use whatever default QR code we choose.

Is this true / is there any plan to support setting a QR code URL from code? I imagine it should be a 1 liner method... I've seen other, unofficial cardboard-ish SDKs support this and it's a really nice feature to have.

**Comment by smdol** [comment]
No, the user's pairing process for viewer+phone is meant to occur outside of any app.  It's a "system" setting that spans all apps, kind of like in Android where you can go into the Settings and change whether the screen orientation is locked or not, or if GPS is enabled.  Apps aren't supposed to do that.

**Comment by brianchasalow** [comment]
We disabled the Cardboard app icloud document sync thing as it was crashing the app in an earlier version of the Cardboard SDK, if a user didn't have the Cardboard app installed. I would imagine this has been fixed by now, but we haven't tested it yet.

It's also kind of strange to have a system setting that has a prerequisite of having another app installed to configure it (on Android, we have to download the Cardboard app/scan it in there to configure it- whereas on iOS we are able to scan it /select a QR code directly in the app.)  It makes onboarding for new users who aren't familiar with VR difficult, when we give out Cardboard at events. "You have to download our app, then download this other app, then scan the code, then go back to our app" 

We're trying to find a way to simplify new (and existing) user experiences. If we're not able to ease the onboarding process by calling a SetQRCode function in our own way, maybe you guys should find a way to improve that device selection in the Cardboard SDK.

**Comment by ezonecom** [comment]
Is there a way to override the default values for the case where the user doesn't have the cardboard app? eg we are working on an app that targets the ViewMaster

Simon

**Comment by nathanmartz** [comment]
On both iOS and Android you can set the default viewer for your app by
modifying the DefaultDevice in GVRViewer.cs. This will be used unless the
user has already paired a device. Assuming you are giving out cardboards at
events, just set that as the default for your app and it'll make the setup
flow very easy for most users at the event.

**Comment by brianchasalow** [comment]
@nathanmartz Sorry for taking awhile to follow up here! We might've met at Google I/O this year- I'm at Within. I just want to say we often give cardboards out at events, but often the percentage of people with an event-cardboard is a minority, compared to the rest of the population using our app who might do better with a more generic default QR code. It might be a bad idea to switch out the default cardboard for an event, when it might affect everyone else all over the world using the app. 

Maybe we could do something clever where we detect a user's location, and if they're near the special event site they get a different 'default' QR code.  That seems like a bit of a messy workaround though... I'm open to suggestions!

**Comment by smdol** [comment]
You can certainly decide programmatically at the app level whether to set the GvrViewer's DefaultDevice to a different profile URL.  It won't affect anyone who has already scanned a code for their own viewer, either.

**Comment by vladut-dascalescu** [comment]
I'm using GVR sdk 1.1 . On GvrViewer I found these two variables : DefaultDeviceProfile and device (of type BaseVRDevice). Which one and how should I change, so the default device will be "VR Box 2" for example (http://www.hypergridbusiness.com/faq/vr-headset-qr-codes/ - where I found the QR code coresponding). I want to make a simple app for this specific cardboard (maybe other with a selector if it can be done easily...). I guess Cardboard application on Android it setting this variable(s) through an activity (in onActivityResult method maybe), or how can those 2 apps communicate ?
